onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp

```diff
@@ -143,15 +143,40 @@ namespace Dml::GraphDescBuilder
         }
     }
 
-    GraphDesc BuildGraphDesc(
+    uint32_t SetAndGetMainDmlGraphNodeIndex(
+        const uint32_t operatorDmlGraphNodeIndex,
+        const std::string& nodeNamePrefix,
+        AbstractOperatorDesc& operatorDesc,
+        /*in_out*/std::unordered_map<uint32_t, uint32_t>& operatorDmlGraphNodeIndexToMainDmlGraphNodeIndexMap,
+        /*in_out*/std::vector<DmlSerializedGraphNode>& dmlGraphNodes)
+    {
+        auto iter = operatorDmlGraphNodeIndexToMainDmlGraphNodeIndexMap.find(operatorDmlGraphNodeIndex);
+        if (iter != operatorDmlGraphNodeIndexToMainDmlGraphNodeIndexMap.end())
+        {
+            return iter->second;
+        }
+        operatorDmlGraphNodeIndexToMainDmlGraphNodeIndexMap[operatorDmlGraphNodeIndex] = static_cast<uint32_t>(dmlGraphNodes.size());
+        dmlGraphNodes.push_back({operatorDesc, nodeNamePrefix + std::to_string(operatorDmlGraphNodeIndex)});
+        return operatorDmlGraphNodeIndexToMainDmlGraphNodeIndexMap[operatorDmlGraphNodeIndex];
+    }
+
+    /*
+    * Terminology:
+    *   SubGraph: partitioned ONNX graph from the original (main) ONNX graph
+    *   DmlGraph: a graph in DML currency converted from subGraph.
+    *   operatorDmlGraph: a graph in DML currency for a given node or operator
+    * DmlGraph aka mainDmlGraph to distinguish b/w operatorDmlGraph and DmlGraph.
```

Comment:
>b/w [](http://example.com/codeflow?start=47&length=3)

```suggest
between
```

(it doesn't save that many characters, and it's kinda confusing because it makes graphics people think of black and white for a head scratching second :b)